internal/resources/providers/gcplib/inventory/provider.go

```diff
@@ -398,7 +398,7 @@ func getAssetsByProject[T any](assets []*ExtendedGcpAsset, log *clog.Logger, f T
 
 func (p *Provider) getAllAssets(ctx context.Context, request *assetpb.ListAssetsRequest) []*assetpb.Asset {
 	p.log.Infof("Listing asset types: %v of type %v for %v", request.AssetTypes, request.ContentType, request.Parent)
-	results := make([]*assetpb.Asset, 0)
+	results := make([]*assetpb.Asset, 0) // TODO: use PageInfo().Remaining() to initialize with capacity <issue link>
```

Comment:
Let's remove this TODO if we don't have an issue for it and if it's not a compatible change with streaming each page anyway